src/io/json/read/infer_schema.rs

```diff
@@ -186,4 +191,15 @@ mod test {
             List(Box::new(Field::new(ITEM_NAME, Utf8, true))),
         );
     }
+
+    #[test]
+    fn test_coersion_of_nulls() {
+        assert_eq!(coerce_data_type(&[DataType::Null]), DataType::Null);
+        assert_eq!(
+            coerce_data_type(&[DataType::Null, DataType::Boolean]),
+            DataType::Utf8
+        );
```

Comment:
Is this intended? Should it not be just `DataType::Boolean`?

Similarly, `{"a": null}"` is currently infered as `DataType::Struct(vec![])`and not `DataType::Struct(vec![Field::new("a", DataType::Null, true)])`

Reply:
Good observation. I think it is intended - `infer_array` does not pass `DataType::Null` to `coerce_data_type` since arrow arrays are nullable and thus we handle `Null` as part before passing to `coerce_data_type`. The idea is [here](https://github.com/jorgecarleitao/arrow2/blob/main/src/io/json/read/infer_schema.rs#L56).

With that said, imo it is not the cleanest design and maybe we should clean it up?